### PR TITLE
Feature: Allow admin to change the navbar color, solves #39, helps to resolve #110

### LINKED ISSUE #110: Feature: Custom color-branded navbar
Extends #39 

I started work on this but stumbled into some problems, some can be resolved others have some further impact.

**Resolved:**
Automatically choose the right font-color depending on the contrast, etc.

**Needs discussion:**
The navbar background color might interfere with the `brandcolor` therefore the navigation highlighting of the currently active tab and the edit button might not be visible good enough. I am not sure anymore how much this setting can be generalized in a meaningful way.

### PR DESCRIPTION
**Pull request by abias**
This replaces the PR #84 which was reviewed by me but where I decided to put some more effort in it and extend the contributed functionality.

It allows the admin to switch to a dark navbar:
grafik

and it also allows the admin to switch to a navbar which is colored with the brand color:
grafik

I will ask my designer colleagues if they can tweak the dark and colored navbar even more and submit the tweaks as PR, but this is just a matter of SCSS improvements and won't change the functionality.

### CODE REVIEW

**Review by lucaboesch — DISMISSED** [review]
Unfortunately, it is not that beautiful when you have the same identical string twice in the settings, and once where you would expect additional explanations.
image
Please refine this patch regarding this admittedly very small detail which is visible to admins only anyway.

File: settings.php [hunk 1]
```diff
@@ -389,6 +389,29 @@
...
+        // Create navbar heading.
+        $name = 'theme_boost_union/navbarheading';
+        $title = get_string('navbarheading', 'theme_boost_union', null, true);
+        $setting = new admin_setting_heading($name, $title, null);
+        $tab->add($setting);
+
+        // Setting: Navbar color.
+        $name = 'theme_boost_union/navbarcolor';
+        $title = get_string('navbarcolorsetting', 'theme_boost_union', null, true);
+        $description = get_string('navbarcolorsetting', 'theme_boost_union', null, true);
```
Comment (lucaboesch) [inline settings.php:401]: That should be navbarcolorsetting_desc, probably.

**Comment by abias** [comment]
Well spotted, @lucaboesch !
I have indeed missed the `_desc` part in the string identifier!

**Comment by abias** [comment]
Everything is solved and I have merged this now.

**Comment by Amrita1991** [comment]
Hi @abias ,
I see changes related to this issue has been pushed to master branch but there has been no new release. We use a fork of the repo  which gets updated when there is a new release. We want to test this feature and decide on default colours for our university. Hence we were wondering are you planning for a new release soon? if yes, then when?

**Comment by abias** [comment]
@Amrita1991 - thanks for asking.

Our release schedule is currently like this: Every weekend, if there were enough features / changes integrated since the last release, we backport the patches from master into the stable branches, tag a new release in Github and release the new version on moodle.org.

Thus, please expect the next release to be published by the beginning of next week.

**Pull request merged**